From stable-bounces@linux.kernel.org  Wed Jun 29 18:12:57 2005
To: stable@kernel.org
From: akpm@osdl.org
Date: Wed, 29 Jun 2005 18:12:05 -0700
Cc: akpm@osdl.org, kambarov@berkeley.edu, alan@lxorguk.ukuu.org.uk,
        zkambarov@coverity.com
Subject: [patch] coverity: tty_ldisc_ref return null check

From: KAMBAROV, ZAUR <kambarov@berkeley.edu>

We add a check of the return value of tty_ldisc_ref(), which
is checked 7 out of 8 times, e.g.:

149  		ld = tty_ldisc_ref(tty);
150  		if (ld != NULL) {
151  			if (ld->set_termios)
152  				(ld->set_termios)(tty, &old_termios);
153  			tty_ldisc_deref(ld);
154  		}

This defect was found automatically by Coverity Prevent, a static analysis
tool.

(akpm: presumably `ld' is never NULL.  Oh well)

Signed-off-by: Zaur Kambarov <zkambarov@coverity.com>
Acked-by: Alan Cox <alan@lxorguk.ukuu.org.uk>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Chris Wright <chrisw@osdl.org>
---

 drivers/char/tty_ioctl.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff -puN drivers/char/tty_ioctl.c~coverity-tty_ldisc_ref-return-null-check drivers/char/tty_ioctl.c
--- stable/drivers/char/tty_ioctl.c~coverity-tty_ldisc_ref-return-null-check	2005-06-29 18:11:16.000000000 -0700
+++ stable-akpm/drivers/char/tty_ioctl.c	2005-06-29 18:11:16.000000000 -0700
@@ -476,11 +476,11 @@ int n_tty_ioctl(struct tty_struct * tty,
 			ld = tty_ldisc_ref(tty);
 			switch (arg) {
 			case TCIFLUSH:
-				if (ld->flush_buffer)
+				if (ld && ld->flush_buffer)
 					ld->flush_buffer(tty);
 				break;
 			case TCIOFLUSH:
-				if (ld->flush_buffer)
+				if (ld && ld->flush_buffer)
 					ld->flush_buffer(tty);
 				/* fall through */
 			case TCOFLUSH:
_
